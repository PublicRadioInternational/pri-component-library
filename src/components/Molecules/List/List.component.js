/**
 * @file List.component.js
 * Exports a simple list component.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './List.css';

const cx = classNames.bind(styles);

/**
 * Component that renders a list.
 */
export default class List extends Component {
  static propTypes = {
    listItems: PropTypes.arrayOf(PropTypes.object),
    className: PropTypes.string,
    ulClass: PropTypes.string,
    liClass: PropTypes.string,
    linkClass: PropTypes.string,
    role: PropTypes.string,
    ariaLabelledby: PropTypes.string,
    reveal: PropTypes.bool,
    classNameOpen: PropTypes.string,
    title: PropTypes.string,
    titleClass: PropTypes.string
  };

  static defaultProps = {
    listItems: {},
    className: null,
    ulClass: null,
    liClass: null,
    linkClass: null,
    role: null,
    ariaLabelledby: null,
    reveal: false,
    classNameOpen: null,
    title: null,
    titleClass: null
  };

  render() {
    const {
      listItems,
      className,
      role,
      ariaLabelledby,
      ulClass,
      liClass,
      linkClass,
      reveal,
      classNameOpen,
      title,
      titleClass
    } = this.props;

    const listWrapClasses = cx({
      [className]: className,
      [classNameOpen]: reveal === true
    });

    const linkClasses = cx({
      listLink: true,
      [linkClass]: linkClass
    });

    const items = listItems.map(item => (
      <li className={`${styles.listItem} ${liClass}`} key={item.name}>
        <a
          className={`${linkClasses} ${
            item.itemLinkClass !== undefined ? styles[item.itemLinkClass] : ''
          }`}
          href={item.url}
        >
          {item.name}
        </a>
      </li>
    ));
    return (
      <div
        className={listWrapClasses}
        role={role}
        aria-labelledby={ariaLabelledby}
      >
        {title && <h2 className={titleClass}>{title}</h2>}
        <ul className={`${styles.list} ${ulClass}`}>{items}</ul>
      </div>
    );
  }
}
