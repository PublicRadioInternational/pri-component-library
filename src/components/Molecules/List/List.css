/* import colors... */
@value colors: "../../00_global/colors.css";
@value grayLight, orange from colors;

.list {
  margin: 6px 0;
  padding: 0;
}

.listItem {
  display: inline;
  padding: 0 0 0 6px;
  text-align: center;
}

.listItem::after {
  color: grayLight;
  content: ' | ';
  font-size: 13px;
}

.listItem:last-child::after {
  content: '';
}

.listLink {
  color: grayLight;
  font-size: 13px;
  line-height: 2;
  padding-right: 6px;
  text-decoration: none;
}

.listLink:hover {
  color: orange;
}
