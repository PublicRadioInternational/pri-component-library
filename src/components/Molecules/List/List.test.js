/**
 * @file List.test.js
 * Contains tests for List.component.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import List from './List.component';

describe('<List />', () => {
  it('Matches the List snapshot', () => {
    const component = renderer
      .create(
        <List
          listItems={[
            { name: 'Google', url: 'http://google.com' },
            { name: 'Yahoo', url: 'http://yahoo.com' }
          ]}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
