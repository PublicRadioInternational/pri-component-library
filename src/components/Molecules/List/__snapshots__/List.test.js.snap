// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> Matches the List snapshot 1`] = `
<div
  aria-labelledby={null}
  className=""
  role={null}
>
  <ul
    className="list null"
  >
    <li
      className="listItem null"
    >
      <a
        className="listLink undefined"
        href="http://google.com"
      >
        Google
      </a>
    </li>
    <li
      className="listItem null"
    >
      <a
        className="listLink undefined"
        href="http://yahoo.com"
      >
        Yahoo
      </a>
    </li>
  </ul>
</div>
`;
